demos/greengrass_connectivity/aws_greengrass_discovery_demo.c

```diff
@@ -58,18 +58,78 @@
 #include "aws_greengrass_discovery.h"
 
 /* Includes for initialization. */
-#include "iot_mqtt.h"
+#include "core_mqtt.h"
 
-#define ggdDEMO_MAX_MQTT_MESSAGES              3
-#define ggdDEMO_MAX_MQTT_MSG_SIZE              500
-#define ggdDEMO_DISCOVERY_FILE_SIZE            2500
-#define ggdDEMO_MQTT_MSG_TOPIC                 "freertos/demos/ggd"
-#define ggdDEMO_MQTT_MSG_DISCOVERY             "{\"message\":\"Hello #%lu from FreeRTOS to Greengrass Core.\"}"
+/* Include header for connection configurations. */
+#include "aws_clientcredential.h"
 
-#define ggdDEMO_KEEP_ALIVE_INTERVAL_SECONDS    1200
+/* Include header for client credentials. */
+#include "aws_clientcredential_keys.h"
+
+/* Include PKCS11 helpers header. */
+#include "pkcs11_helpers.h"
+
+/* Include AWS IoT metrics macros header. */
+#include "aws_iot_metrics.h"
+
+/* Include header for root CA certificates. */
+#include "iot_default_root_certificates.h"
+
+/* Transport interface implementation include header for TLS. */
+#include "transport_secure_sockets.h"
+
+/* Retry utilities include. */
+#include "backoff_algorithm.h"
+
+#define ggdDEMO_MAX_MQTT_MESSAGES                    3
+#define ggdDEMO_MAX_MQTT_MSG_SIZE                    500
+#define ggdDEMO_DISCOVERY_FILE_SIZE                  2500
+#define ggdDEMO_MQTT_MSG_TOPIC                       "freertos/demos/ggd"
+#define ggdDEMO_MQTT_MSG_DISCOVERY                   "{\"message\":\"Hello #%lu from FreeRTOS to Greengrass Core.\"}"
+
+#define ggdDEMO_KEEP_ALIVE_INTERVAL_SECONDS          1200
 /* Number of times to try MQTT connection. */
-#define ggdDEMO_NUM_TRIES                      3
-#define ggdDEMO_RETRY_WAIT_MS                  2000
+#define ggdDEMO_NUM_TRIES                            3
+#define ggdDEMO_RETRY_WAIT_MS                        2000
+
+#define NETWORK_BUFFER_SIZE                          ( 1024U )
+
+/**
+ * @brief Milliseconds per second.
+ */
+#define MILLISECONDS_PER_SECOND                      ( 1000U )
+
+/**
+ * @brief Milliseconds per FreeRTOS tick.
+ */
+#define MILLISECONDS_PER_TICK                        ( MILLISECONDS_PER_SECOND / configTICK_RATE_HZ )
+
+/**
+ * @brief The maximum number of retries for network operation with server.
+ */
+#define RETRY_MAX_ATTEMPTS                           ( 5U )
+
+/**
+ * @brief The maximum back-off delay (in milliseconds) for retrying failed operation
+ *  with server.
+ */
+#define RETRY_MAX_BACKOFF_DELAY_MS                   ( 5000U )
+
+/**
+ * @brief The base back-off delay (in milliseconds) to use for network operation retry
+ * attempts.
+ */
+#define RETRY_BACKOFF_BASE_MS                        ( 500U )
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define mqttexampleCONNACK_RECV_TIMEOUT_MS           ( 1000U )
```

Comment:
The other macros either use `ggdDEMO_` or just omit a prefix, so that should probably be done here too since this isn't an MQTT demo